src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -293,6 +293,39 @@ public override bool HasDefaultValue
             return defaultValue;
         }
 
+        private object? GetDefaultValueFromCustomAttributeData()
+        {
+            foreach (CustomAttributeData attributeData in RuntimeCustomAttributeData.GetCustomAttributes(this))
+            {
+                Type attributeType = attributeData.AttributeType;
+                if (attributeType == typeof(DecimalConstantAttribute))
+                {
+                    return GetRawDecimalConstant(attributeData);
+                }
+                else if (attributeType.IsSubclassOf(typeof(CustomConstantAttribute)))
+                {
+                    if (attributeType == typeof(DateTimeConstantAttribute))
+                    {
+                        return GetRawDateTimeConstant(attributeData);
+                    }
+                    return GetRawConstant(attributeData);
+                }
+            }
+            return DBNull.Value;
+        }
+
+        private object? GetDefaultValueFromCustomAttributes()
+        {
+            foreach (object customAttribute in GetCustomAttributes(false))
```

Comment:
`GetCustomAttributes(false)` can instantiate a lot of unrelated custom attributes. It is potentially significantly more expensive and it can have unexpected side-effects. People do all sorts of weird things with custom attributes.

I understand that you have done this for consistency with `GetDefaultValueFromCustomAttributeData`, but it may be better to stick with `GetCustomAttributes(typeof(CustomConstantAttribute), false)` + `GetCustomAttributes(typeof(DecimalConstantAttribute), false)`, and live with the subtle difference.